src/main/java/com/example/securitynew/repository/BookRepository.java

```diff
@@ -7,5 +7,4 @@
 
 public interface BookRepository extends JpaRepository<Book, Long>,
         JpaSpecificationExecutor<Book>, PagingAndSortingRepository<Book, Long> {
```

Comment:
YOu dont need jpaspecification and pagingandsorting, cause jpa repository contains already it.
```suggestion
        JpaSpecificationExecutor<Book>, PagingAndSortingRepository<Book, Long> {
```